### DISCUSSION TITLE
`CacheMissHandler` doesn't work as expected

### DISCUSSION THREAD

**Discussion by lukasbalaz** [serverpod/serverpod#2951] [Q&A] [answered]
I am putting this in Q&A instead of creating an issue because there is high chance there is some good reason it works like this but for me it's just really unintuitive.

If I use something like this:
```dart
final userData = session.caches.localPrio.get(
  'User-$id',
  CacheMissHandler(
    () async {
      User user = await getUserFromDb(id);
      return user;
    },
    lifetime: const Duration(minutes: 10),
  ),
);
```
I would expect `userData` never to be `null`. Either it gets the data from cache or from the database.

But for some reason it actually returns `null` if the entry exists in the cache but is expired (see line 91 below). The `cacheMissHandler` is only invoked on a subsequent call.
https://github.com/serverpod/serverpod/blob/main/packages/serverpod/lib/src/cache/local_cache.dart#L88-L92

For me it looks weird and unexpected. Is there some good reason for this?

**Comment by Isakdl** [comment, marked as answer]
Your confusion makes sense, this is a bug! Opening a ticket for that, thank you!
